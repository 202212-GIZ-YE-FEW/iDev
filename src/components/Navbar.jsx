import clsx from "clsx";
import Image from "next/image";
import Link from "next/link";
import { usePathname } from "next/navigation";
import { withTranslation } from "next-i18next";
import { useEffect, useState } from "react";

import Dropdown from "@/components/Dropdown";
import Button from "@/components/ui/Button";

import { authenticatedDropdown, langs, navigation } from "@/utils/constants";

import { useAuth } from "./context/AuthContext";

function NavLink({ to, children }) {
    return (
        <Link
            href={to}
            className={clsx(
                "my-4 lg:my-0 lg:mx-2 xl:mx-5 text-md lg:text-md xl:text-lg relative hover:text-yellow",
                {
                    "text-yellow font-semibold underline": usePathname() === to,
                }
            )}
        >
            {children}
        </Link>
    );
}
function MobileNav(prop) {
    const { openHamburger, setActiveDropdown, activeDropdown, i18n, path, t } =
        prop;
    const { authenticated } = useAuth();
    return (
        <div
            className={`lg:hidden absolute top-0 right-0 h-full w-full z-10 bg-background transform ${
                openHamburger
                    ? "ltr:translate-x-0 rtl:-translate-x-0"
                    : "hidden"
            } transition-transform duration-300 ease-in-out`}
        >
            <div className='container flex items-center bg-light-cyan h-20'>
                {/* Logo */}
                <div className='flex items-center'>
                    <Link
                        className='text-2xl font-semibold flex items-center space-s-4'
                        href='/'
                    >
                        <Image
                            priority
                            src='/logo.svg'
                            alt='Healing Logo'
                            width={30}
                            height={30}
                        />
                        <span className='text-2xl md:text-3xl lg:text-4xl xl:text-5xl uppercase'>
                            Healing
                        </span>
                    </Link>
                </div>
            </div>
            {/* Dropdown panel */}
            <div className='container flex flex-col bg-background'>
                {/* Navigation links */}
                {navigation.map((nav) => {
                    return nav.name === "about" ? (
                        <Dropdown
                            activeDropdown={activeDropdown}
                            links={nav.links}
                            item='about'
                            mobile={true}
                            setActiveDropdown={setActiveDropdown}
                            icon={
                                <button className='flex items-center'>
                                    <Image
                                        priority
                                        src='/arrow.svg'
                                        width={15}
                                        height={15}
                                        alt=''
                                    />
                                    <span className='inline-block ms-1'>
                                        {t(`${nav.name}`)}
                                    </span>
                                </button>
                            }
                            t={t}
                        />
                    ) : (
                        <NavLink key={nav.name} to={nav.href}>
                            {t(`${nav.name}`)}
                        </NavLink>
                    );
                })}
                <div className='flex space-s-10'>
                    {/* Langs Dropdown */}
                    <Dropdown
                        activeDropdown={activeDropdown}
                        links={langs}
                        mobile={true}
                        item='langs'
                        to={path}
                        i18n={i18n}
                        setActiveDropdown={setActiveDropdown}
                        icon={
                            <button className='p-2 text-black bg-yellow rounded-md focus:outline-none lg:mx-3'>
                                <Image
                                    priority
                                    src='/lang.svg'
                                    width={23}
                                    height={23}
                                    alt=''
                                />
                            </button>
                        }
                    />
                    {!authenticated && (
                        <Link href='/login'>
                            <Button
                                content={t("login")}
                                textTransform='capitalize'
                                filled='true'
                                size='large'
                                fontSize='text-sm md:text-xl'
                                radius='md'
                            />
                        </Link>
                    )}
                </div>
            </div>
        </div>
    );
}

function Navbar({ t, i18n }) {
    const path = usePathname() || "/";

    const [openHamburger, setOpenHamburger] = useState(false);
    const [activeDropdown, setActiveDropdown] = useState(null);
    const { authenticated, user } = useAuth();
    const [photo, uploadimg] = useState("");

    useEffect(() => {
        uploadimg(user.photoURL);
    }, [user]);
    return (
        <nav className='bg-light-cyan'>
            <div className='container flex h-20 py-8 items-center'>
                {/* Logo */}
                <div className='w-4/12 flex items-center'>
                    <Link
                        className='text-2xl font-semibold flex items-center space-s-4'
                        href='/'
                    >
                        <Image
                            priority
                            src='/logo.svg'
                            alt='Healing Logo'
                            width={30}
                            height={30}
                        />
                        <span className='text-2xl md:text-3xl lg:text-4xl xl:text-5xl uppercase'>
                            Healing
                        </span>
                    </Link>
                </div>
                <div className='w-8/12 justify-end flex items-center'>
                    {/* Hamburger button */}
                    <div
                        className='flex relative z-20 order-last w-8 h-5 flex-col justify-between items-center lg:hidden'
                        onClick={() => setOpenHamburger(!openHamburger)}
                    >
                        <span
                            className={`h-1 w-full bg-black rounded-lg transform transition duration-300 ease-in-out ${
                                openHamburger ? "rotate-45 translate-y-1.5" : ""
                            }`}
                        />
                        <span
                            className={`h-1 w-full bg-black rounded-lg transition-all duration-300 ease-in-out ${
                                openHamburger ? "hidden" : "w-full"
                            }`}
                        />
                        <span
                            className={`h-1 w-full bg-black rounded-lg transform transition duration-300 ease-in-out ${
                                openHamburger
                                    ? "-rotate-45 -translate-y-2.5"
                                    : ""
                            }`}
                        />
                    </div>
                    <div className='hidden lg:flex items-center text-end'>
                        {/* Navigation links */}
                        {navigation.map((nav) => {
                            return nav.name === "about" ? (
                                <Dropdown
                                    activeDropdown={activeDropdown}
                                    links={nav.links}
                                    item='about'
                                    setActiveDropdown={setActiveDropdown}
                                    icon={
                                        <button className='flex items-center md:mx-2 xl:mx-6'>
                                            <Image
                                                priority
                                                src='/arrow.svg'
                                                width={15}
                                                height={15}
                                                alt=''
                                            />
                                            <span className='inline-block ms-1'>
                                                {t(`${nav.name}`)}
                                            </span>
                                        </button>
                                    }
                                    t={t}
                                />
                            ) : (
                                <NavLink key={nav.name} to={nav.href}>
                                    {t(`${nav.name}`)}
                                </NavLink>
                            );
                        })}
                        {/* Langs Dropdown */}
                        <Dropdown
                            activeDropdown={activeDropdown}
                            links={langs}
                            to={path}
                            item='langs'
                            i18n={i18n}
                            setActiveDropdown={setActiveDropdown}
                            icon={
                                <button className='p-2 text-black bg-yellow rounded-md focus:outline-none lg:mx-2'>
                                    <Image
                                        priority
                                        src='/lang.svg'
                                        width={23}
                                        height={23}
                                        alt=''
                                    />
                                </button>
                            }
                        />
                        {!authenticated && (
                            <Link href='/login'>
                                <Button
                                    content={t("login")}
                                    text-transform='capitalize'
                                    filled='true'
                                    size='small'
                                    fontSize='text-sm md:text-base'
                                    radius='md'
                                    interaction='transform hover:bg-yellow transition hover:scale-75 active:bg-cyan focus:outline-none focus:ring focus:ring-cyan'
                                />
                            </Link>
                        )}
                    </div>
                    {/* Dropdown for authenticated user */}
                    {authenticated && (
                        <Dropdown
                            activeDropdown={activeDropdown}
                            links={authenticatedDropdown}
                            item='profile'
                            setActiveDropdown={setActiveDropdown}
                            icon={
                                <button className='mx-2 w-9 max-w-14 max-h-14 rounded-full items-center'>
                                    <Image
                                        className='rounded-full w-full h-full object-cover'
                                        width={14}
                                        height={14}
                                        alt='userImage'
                                        src={photo}
                                    />
                                </button>
                            }
                            t={t}
                        />
                    )}
                </div>
            </div>
            <MobileNav
                openHamburger={openHamburger}
                activeDropdown={activeDropdown}
                setActiveDropdown={setActiveDropdown}
                i18n={i18n}
                t={t}
                path={path}
            />
        </nav>
    );
}
export default withTranslation("common")(Navbar);
