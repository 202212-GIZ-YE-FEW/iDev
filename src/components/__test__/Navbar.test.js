import renderer from "react-test-renderer";

import Navbar from "@/components/Navbar";

it("renders correctly", () => {
    const tree = renderer.create(<Navbar />).toJSON();
    expect(tree).toMatchSnapshot();
});
