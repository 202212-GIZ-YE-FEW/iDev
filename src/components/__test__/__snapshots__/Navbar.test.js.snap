// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav
  className="bg-light-cyan"
>
  <div
    className="container flex h-20 py-8 items-center"
  >
    <div
      className="lg:hidden absolute top-0 right-0 h-full w-full z-20 bg-background transform ltr:-translate-x-full rtl:translate-x-full opacity-0 transition-transform duration-300 ease-in-out"
    >
      <div
        className="flex items-center justify-between bg-light-cyan h-20 px-4"
      >
        <a
          className="text-2xl font-semibold flex items-center space-s-4"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <img
            alt="Healing Logo"
            data-nimg="1"
            decoding="async"
            height={30}
            onError={[Function]}
            onLoad={[Function]}
            src="/logo.svg"
            style={
              Object {
                "color": "transparent",
              }
            }
            width={30}
          />
          <span
            className="text-2xl md:text-3xl lg:text-4xl xl:text-5xl uppercase"
          >
            Healing
          </span>
        </a>
      </div>
      <div
        className="flex flex-col px-4"
      >
        <a
          className="my-4 lg:my-0 lg:mx-2 xl:mx-5 text-md lg:text-md xl:text-lg relative hover:text-yellow"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          home
        </a>
        <a
          className="my-4 lg:my-0 lg:mx-2 xl:mx-5 text-md lg:text-md xl:text-lg relative hover:text-yellow"
          href="#recentblog"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          blogs
        </a>
        <a
          className="my-4 lg:my-0 lg:mx-2 xl:mx-5 text-md lg:text-md xl:text-lg relative hover:text-yellow"
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="relative"
            onClick={[Function]}
          >
            <div
              className="flex"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height={15}
                onError={[Function]}
                onLoad={[Function]}
                src="/arrow.svg"
                style={
                  Object {
                    "color": "transparent",
                  }
                }
                width={15}
              />
              <span
                className="inline-block ms-1"
              >
                about
              </span>
            </div>
          </div>
        </a>
        <a
          className="my-4 lg:my-0 lg:mx-2 xl:mx-5 text-md lg:text-md xl:text-lg relative hover:text-yellow"
          href="/contact-us"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          contactUs
        </a>
        <div
          className="flex space-s-10"
        >
          <div
            className="relative inline-block"
            onClick={[Function]}
          >
            <button
              className="relative z-10 block p-2 text-black bg-yellow rounded-md focus:outline-none mx-3"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height={23}
                onError={[Function]}
                onLoad={[Function]}
                src="/lang.svg"
                style={
                  Object {
                    "color": "transparent",
                  }
                }
                width={23}
              />
            </button>
          </div>
          <a
            href="/login"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            <button
              className="transform hover:bg-yellow transition duration-500 hover:animate-bounce hover:scale-75 active:bg-cyan focus:outline-none focus:ring focus:ring-cyan
                font-normal whitespace-nowrap cursor-pointer  capitalize text-sm md:text-xl rounded-md px-[25px] md:px-[35px] py-[8px] font-medium bg-cyan text-black "
              disabled={false}
            >
              login
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      className="w-3/12 flex items-center"
    >
      <a
        className="text-2xl font-semibold flex items-center space-s-4"
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <img
          alt="Healing Logo"
          data-nimg="1"
          decoding="async"
          height={30}
          onError={[Function]}
          onLoad={[Function]}
          src="/logo.svg"
          style={
            Object {
              "color": "transparent",
            }
          }
          width={30}
        />
        <span
          className="text-2xl md:text-3xl lg:text-4xl xl:text-5xl uppercase"
        >
          Healing
        </span>
      </a>
    </div>
    <div
      className="w-9/12 flex justify-end items-center"
    >
      <div
        className="z-50 flex relative w-8 h-5 flex-col justify-between items-center lg:hidden"
        onClick={[Function]}
      >
        <span
          className="h-1 w-full bg-black rounded-lg transform transition duration-300 ease-in-out "
        />
        <span
          className="h-1 w-full bg-black rounded-lg transition-all duration-300 ease-in-out w-full"
        />
        <span
          className="h-1 w-full bg-black rounded-lg transform transition duration-300 ease-in-out "
        />
      </div>
      <div
        className="hidden lg:flex items-center"
      >
        <a
          className="my-4 lg:my-0 lg:mx-2 xl:mx-5 text-md lg:text-md xl:text-lg relative hover:text-yellow"
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          home
        </a>
        <a
          className="my-4 lg:my-0 lg:mx-2 xl:mx-5 text-md lg:text-md xl:text-lg relative hover:text-yellow"
          href="#recentblog"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          blogs
        </a>
        <a
          className="my-4 lg:my-0 lg:mx-2 xl:mx-5 text-md lg:text-md xl:text-lg relative hover:text-yellow"
          href="#"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <div
            className="relative"
            onClick={[Function]}
          >
            <div
              className="flex"
            >
              <img
                alt=""
                data-nimg="1"
                decoding="async"
                height={15}
                onError={[Function]}
                onLoad={[Function]}
                src="/arrow.svg"
                style={
                  Object {
                    "color": "transparent",
                  }
                }
                width={15}
              />
              <span
                className="inline-block ms-1"
              >
                about
              </span>
            </div>
          </div>
        </a>
        <a
          className="my-4 lg:my-0 lg:mx-2 xl:mx-5 text-md lg:text-md xl:text-lg relative hover:text-yellow"
          href="/contact-us"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          contactUs
        </a>
        <div
          className="relative inline-block"
          onClick={[Function]}
        >
          <button
            className="relative z-10 block p-2 text-black bg-yellow rounded-md focus:outline-none mx-3"
          >
            <img
              alt=""
              data-nimg="1"
              decoding="async"
              height={23}
              onError={[Function]}
              onLoad={[Function]}
              src="/lang.svg"
              style={
                Object {
                  "color": "transparent",
                }
              }
              width={23}
            />
          </button>
        </div>
        <a
          href="/login"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          <button
            className="transform hover:bg-yellow transition hover:scale-75 active:bg-cyan focus:outline-none focus:ring focus:ring-cyan
                font-normal whitespace-nowrap cursor-pointer  capitalize text-sm md:text-xl rounded-md px-[25px] md:px-[35px] py-[8px] font-medium bg-cyan text-black "
            disabled={false}
          >
            login
          </button>
        </a>
      </div>
    </div>
  </div>
</nav>
`;
